### DISCUSSION TITLE
浏览器适配问题

### DISCUSSION THREAD

**Discussion by ahsdhajshqwje** [Tencent/libpag#1654] [General] [answered]
使用edge播放pag动画正常显示，浏览器版本：92.0.902.84
但在google上播放自动会把pag动画放大四倍，导致错位下方是渲染代码，这里给它宽高固定为 3200，1120 但是通过查看canvas标签发现宽高全都被放大了四倍

 const PAG = await window.libpag.PAGInit();
      let url = "/pag/" + this.url + ".pag"
      let id = this.renderingId
      const buffer = await fetch(url).then((response) => {
        return response.arrayBuffer()
      })

      const pagFile = await PAG.PAGFile.load(buffer)
      const pagComposition = PAG.PAGComposition.make( 3200, 1120 );
       pagComposition.addLayer(pagFile);
      const canvas = document.getElementById(id);
      if (canvas !== null) {
        canvas.width = 3200
        canvas.height = 1120
         const pagView = await PAG.PAGView.init(pagComposition, canvas );

        pagView.setRepeatCount(0);
        await pagView.play();
     }

**Comment by zhalice2011** [comment, marked as answer]
你好, 关于 PAG 自适应请看下面的文档和示例代码

文档: https://pag.art/docs/web-play-pagfile.html

示例: PAG的填充模式和 css的cover contain 是不完全对等的, PAG填充模式的实现是先缩放然后居中

不过你可以不使用 setScaleMode , 而去使用手动设置 Matrix 来实现和css中一致的效果

预览: https://pagtest.zhalice.com/pages/scale-mode-background-after.html

源代码: view-source:https://pagtest.zhalice.com/pages/scale-mode-background-after.html
